src/Device/Driver/FreeVario.cpp

```diff
@@ -0,0 +1,291 @@
+                   if (newmc <= 0){newmc = 0.0;}
+                     info.settings.ProvideMacCready(newmc,info.clock);
+               }
+               break;
+             }
+             case 'B': {
+                   if (subCommand == 'S'){
+
+                                double bugsIn;
+                                bool bugsOK = line.ReadChecked(bugsIn);
+
+                                if (subCommand == 'S' && bugsOK){
+                                  sprintf(buffer,"PFV,OK,%s","OK BUG IN value");
+                                  PortWriteNMEA(port, buffer, env);
+                                  info.settings.ProvideBugs((100 -bugsIn )/100.,info.clock);
+                                }
+                                else {
+                                  sprintf(buffer,"PFV,ERR,%s","Error BUG IN value");
+                                  PortWriteNMEA(port, buffer, env);
+                                }
+                            }
+                   break;
+               }
+
+             case 'Q': {
+                          if (subCommand == 'S'){
+                          double qnhIn;
+                          bool qnhOK = line.ReadChecked(qnhIn);
+
+                           if (subCommand == 'S' && qnhOK){
+                             // Ballast is setting in percent by capacity
+                             AtmosphericPressure pres = info.static_pressure.HectoPascal(qnhIn);
+                             info.settings.ProvideQNH(pres, info.clock);
+                            }
+                             else {
+                                sprintf(buffer,"PFV,ERR,%s","Error Weight IN value");
+                                 PortWriteNMEA(port, buffer, env);
+                              }
+                           }
+                                     break;
+                            }
+
+             case 'F': {
+               if (subCommand == 'S'){
+                   info.switch_state.flight_mode = SwitchState::FlightMode::CRUISE;
+               }
+               else if (subCommand == 'C'){
+                 info.switch_state.flight_mode = SwitchState::FlightMode::CIRCLING;
+               }
+               break;
+             }
+           }
+       }
+  return true;
+ }
+
+bool
+FreeVarioDevice::ParseNMEA(const char *_line, NMEAInfo &info)
+{
+ NullOperationEnvironment env;
+ char buffer[80];
+
+  if (!VerifyNMEAChecksum(_line)){
+      sprintf(buffer,"PFV,ERR,%s%s", "Checksum:",_line);
+      PortWriteNMEA(port, buffer, env);
+     return false;
+   } else  {
+       NMEAInputLine line(_line);
+       if (line.ReadCompare("$PFV")){
+           return PFVParser(line, info, port);
+       } else {
+           sprintf(buffer,"PFV,ERR,%s", "Wrong Class");
+           PortWriteNMEA(port, buffer, env);
+           return false;
+         }
+   }
+
+}
+
+
+
+void
+FreeVarioDevice::OnCalculatedUpdate(const MoreData &basic,const DerivedInfo &calculated)
```

Comment:
This method sends all values on every CalculationThread step - wouldn't it be better to send only modified values?